lib/include/jxl/encode.h

```diff
@@ -361,14 +363,64 @@ JXL_EXPORT JxlEncoderStatus JxlEncoderProcessOutput(JxlEncoder* enc,
  * JxlEncoderStoreJPEGMetadata and a single JPEG frame is added, it will be
  * possible to losslessly reconstruct the JPEG codestream.
  *
- * @param options set of encoder options to use when encoding the frame.
+ * @param frame_options set of options and metadata for this frame. Also
+ * includes reference to the encoder object.
  * @param buffer bytes to read JPEG from. Owned by the caller and its contents
  * are copied internally.
  * @param size size of buffer in bytes.
  * @return JXL_ENC_SUCCESS on success, JXL_ENC_ERROR on error
  */
 JXL_EXPORT JxlEncoderStatus JxlEncoderAddJPEGFrame(
-    const JxlEncoderOptions* options, const uint8_t* buffer, size_t size);
+    const JxlEncoderOptions* frame_options, const uint8_t* buffer, size_t size);
+
+/**
+ * Sets the frame header with animation information for this frame.
+ *
+ * It is not required to use this function, however if have_animation is set
+ * to true in the basic info, then this function should be used to set the
+ * time duration of this individual frame. By default individual frames have a
+ * time duration of 0, making them form a composite still. See @ref
+ * JxlFrameHeader for more information.
+ *
+ * This information is stored in the JxlEncoderOptions and so is used for any
+ * frame encoded with these JxlEncoderOptions. It is ok to change between @ref
+ * JxlEncoderAddImageFrame calls, each added image frame will have the frame
+ * header that was set in the options at the time of calling
+ * JxlEncoderAddImageFrame.
+ *
+ * The is_last and name_length fields of the JxlFrameHeader are ignored, use
+ * @ref JxlEncoderCloseFrames to indicate last frame, and @ref
+ * JxlEncoderSetFrameName to indicate the name and its length instead. Calling
+ * this function will clear any name that was previously set with @ref
+ * JxlEncoderSetFrameName.
+ *
+ * @param frame_options set of options and metadata for this frame. Also
+ * includes reference to the encoder object.
+ * @param frame_header frame header data to set.
+ * @return JXL_ENC_SUCCESS on success, JXL_ENC_ERROR on error
+ */
+JXL_EXPORT JxlEncoderStatus JxlEncoderSetFrameHeader(
```

Comment:
See my comment above. It might be better to call this `JxlEncoderSetFrameInfo` or something like that, to leave open the possibility that what you're setting here is not necessarily exactly what will be written in the bitstream frameheader, but rather is how you will present the frame to the encoder, which can then choose to encode it just like that, or maybe to make some changes like trimming unchanged parts.

Reply:
Makes sense, renamed to JxlEncoderSetFrameInfo